src/plugins/discover/public/embeddable/saved_search_grid.tsx

```diff
@@ -30,7 +29,7 @@ export function DiscoverGridEmbeddable(props: DiscoverGridEmbeddableProps) {
       responsive={false}
       data-test-subj="embeddedSavedSearchDocTable"
     >
-      {props.totalHitCount !== 0 && (
+      {Boolean(props.totalHitCount) && props.totalHitCount !== 0 && (
```

Comment:
When the contents of the grid are still being loaded, `totalHitCount` is undefined which caused an error in the `TotalDocuments` component because the `FormattedNumber` component **does not** expect an undefined value.